core/controllers/download.py

```diff
@@ -118,3 +111,17 @@ def sort_output_dataframes(df: pd.DataFrame, sheet: str) -> pd.DataFrame:
     df.reset_index(drop=True, inplace=True)
 
     return df
+
+
+def date_to_string(obj) -> str:
+    """Converts a datetime object suitable for JSON serialisation.
+
+    :param obj: A datetime object (or date) that want to serialise into a string.
```

Comment:
"A datetime object (or date) to be serialised into a string."?

Reply:
I have a few gripes with this function in general (the author refused to address them on PR). But basically:

- The function docstring has specific use case in it, but really, this function has a more general purpose as a black-box for casting datetime objects to string. There could be a case for changing the docstring to be more generic, as it could feasibly be re-sued outside the current use case of extending the built-in JSON serialiser.
- The function itself feels more like a general util than anything download specific, maybe move it to utils module?

Reply:
Also, that param isn't specifically typed to only take date or datetime object, I wonder if we should either update the type hint, or at least make the docstring param clear that it takes any object, and that it will attempt to cast to a str rep of date, otherwise raise

Reply:
probably the latter

Reply:
I didn't really read this function, just moved it as I thought the module made more sense with the main `download` function at the top.

Agree with you though - I'll make some changes in line with your points